tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+);
+
+CREATE TABLE IF NOT EXISTS RaceTypes (
+	type_id			UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	hit_die			TEXT
+		NOT NULL,
+	bab_per_hit_die		REAL
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXiSTS RaceSubtypes (
+	subtype_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL
+);
+	
+--Class tables
+CREATE TABLE IF NOT EXISTS Classes (
+	class_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	hit_die			TEXT
+		NOT NULL,
+	starting_wealth		TEXT
+		NOT NULL,
+	bab_per_level		REAL
+		NOT NULL,
+	skills_per_level	INT
+		NOT NULL,
+	skills_attr		Attribute
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Subclasses (
+	subclass_id		UUID PRIMARY KEY,
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+
+	caster_type		CasterType,
+	casting_attr		Attribute
+);
+
+CREATE TABLE IF NOT EXISTS characterSubclasses (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	subclass_id		UUID REFERENCES Subclasses
+		NOT NULL,
+
+	levels_taken		INT
+		NOT NULL
+		CHECK (levels_taken > 0),
+	hp_taken		INT
+		NOT NULL
+		CHECK (hp_taken >= 0),
+	skills_taken		INT
+		NOT NULL
+		CHECK (skills_taken >= 0)
+);
+
+--Feats tables
+CREATE TABLE IF NOT EXISTS Feats (
+	feat_id			UUID PRIMARY KEY,
+
+	short_description	TEXT,
+	long_description	TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS SkillReqUnits (
+	feat_id			UUID REFERENCES Feats
+		NOT NULL,
+	skill_unit_id		UUID REFERENCES SkillFeatUnits
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS SkillFeatUnits (
+	skill_unit_id		UUID PRIMARY KEY,
+	req_skill		Skill
+		NOT NULL,
+	ranks			INT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS AttributeReqUnits (
+	feat_id			UUID REFERENCES Feats
+		NOT NULL,
+	attr_unit_id		UUID REFERENCES AttributeFeatUnits
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS AttributeFeatUnits (
+	attr_unit_id		UUID PRIMARY KEY,
+	req_attr		Attribute
+		NOT NULL,
+	score			INT
+		NOT NULL
+);
+
+--this table creates a many to many realationship internally between feats.
+--any given feat can require a number of other feats to be taken,
+--and a given feat can be required by any number of other feats.
+--this table accomplishes that.
+CREATE TABLE IF NOT EXISTS FeatRequirements (
+	feat_id			UUID REFERENCES Feats
+		NOT NULL,
+	required_feat		UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterFeats (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS RacialFeats (
+	race_id			UUID REFERENCES Races
+		NOT NULL,
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeats (
+	class_id		UUID REFERENCES Classes,
+		NOT NULL
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+--Features tables
+CREATE TABLE IF NOT EXISTS Features (
+	feature_id		UUID PRIMARY KEY,
+	description		TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeatures (
+	class_id		UUID REFERENCES Feats
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL,
+	is_default		BOOLEAN
```

Comment:
For this and other `BOOLEAN` columns, you should define a default value to each. You may leave `NOT NULL` if a column with a default value can still somehow be `NULL`.

Reply:
Implemented.